src/clj/rems/poller/email.clj

```diff
@@ -260,21 +254,28 @@
       (log/info "sending email:" (pr-str email))
       (cond
         to-error
-        (log/warn "failed address validation, skipping message: " to-error)
+        (do
+          (log/warn "failed address validation:" to-error)
+          (str "failed address validation: " to-error))
 
         (not (and host port))
-        (log/info "no smtp server configured, only pretending to send email")
+        (do
+          (log/info "no smtp server configured, only pretending to send email")
+          nil)
 
         :else
         (try
           (postal/send-message {:host host :port port} email)
-          (catch javax.mail.SendFailedException e ; e.g. email address does not exist
-            (log/warn e "failed sending email, skipping:" (pr-str email))))))))
+          nil
+          (catch Throwable e ; e.g. email address does not exist
+            (log/warn e "failed sending email:" (pr-str email))
+            (str "failed sending email: " e)))))))
 
 (defn run []
-  (common/run-event-poller ::poller (fn [event]
-                                      (doseq [mail (event-to-emails event)]
-                                        (send-email! mail)))))
+  (doseq [email (email-outbox/get-emails {:remaining-attempts? true})]
+    (if-let [error (send-email! (:email-outbox/email email))]
+      (email-outbox/attempt-failed! (:email-outbox/id email) error)
+      (email-outbox/attempt-succeeded! (:email-outbox/id email)))))
```

Comment:
could log remaining attempts, or at least when sending failed for the final time